### PR TITLE
Engineering Beginnings

### PR DESCRIPTION
**Pull request by AtnerNT**
Adds the Fusion and Geothermal generators, Fuel cell recyclers, High voltage machine connectors and blow torches.
All behave the same as in TGMC but geothermal generators have no xenomorph interaction yet.
I also mapped the generators and recyclers onto teststation so show how they should be set up.
Gonna work on Requsitions next, basically just purging the cargo shuttle, and then setting up a new test ship & planet scene but ill put those into their own PRs.

(This Pr also changes the SMES sprites to their TGMC ones)

Tests will most likely fail first try because i added a bunch of sprite SOs and not sure if i remebered to add them all to catalogue

### Changelog:
CL: [New] Adds Fusion and Geothermal generators
CL: [New] Adds Fusion cell recyclers
CL: [New] Adds HV machine connector
CL: [New] Adds Blow torches
CL: [Improvement] Changes SMES sprites into TGMC sprites

### CODE REVIEW

File: UnityProject/Assets/Scripts/Core/Input System/InteractionV2/Validations.cs [hunk 1]
```diff
@@ -750,6 +750,19 @@ public static bool HasUsedActiveWelder(Interaction interaction)
...
+	/// </summary>
+	/// <param name="objectToCheck"></param>
+	/// <returns></returns>
+	public static bool HasUsedActiveWelder(GameObject objectToCheck)
+	{
+		if (objectToCheck == null) return false;
+		var welder = objectToCheck.GetComponent<FireSource>();
+		if (welder == null) return false;
+		return welder.IsBurning && HasItemTrait(objectToCheck, CommonTraits.Instance.Welder);
+	}
```
Comment (ThatDan123) [inline UnityProject/Assets/Scripts/Core/Input System/InteractionV2/Validations.cs]: Doing it this way means that blow torches can't be used for anything welders do currently.

I think the preferred solution would be to make the Welder script a base class in the main repo and then inherit two scripts which do the sprites the different ways?

Comment (AtnerNT) [reply UnityProject/Assets/Scripts/Core/Input System/InteractionV2/Validations.cs]: Yeah no this is a far better implementation, ive made the change

**Comment by AtnerNT** [comment]
I've also done the ASRS fix in this PR to save merge conflicts and the like, so Im going to close that PR

**Comment by AtnerNT** [comment]
Cargo tests failing but that's not this PR, should be good to merge

**Pull request merged**